os/include/tinyara/fs/automount.h

```diff
@@ -0,0 +1,209 @@
+ * automounter.  This structure provides information about the volume to be
+ * mounted and provides board-specific hooks.
+ *
+ * Memory for this structure is provided by the caller.  It is not copied
+ * by the automounter and is presumed to persist while the automounter
+ * is active.
+ */
+
+struct automount_lower_s {
+	/* Volume characterization */
+
+	FAR const char *fstype;     /* Type of file system */
+	FAR const char *blockdev;   /* Path to the block device */
+	FAR const char *mountpoint; /* Location to mount the volume */
+
+	/* Debounce delay in system clock ticks.  Automount operation will not
+	 * be performed until the insertion/removal state has been unchanges
+	 * for this duration.
+	 */
+
+	uint32_t ddelay;
+
+	/* Unmount delay time in system clock ticks.  If a volume has open
+	 * references at the time that the media is removed, then we will be
+	 * unable to unmount it.  In that case, hopefully, the clients of the
+	 * mount will eventually fail with file access errors and eventually close
+	 * their references.  So, at some time later, it should be possible to
+	 * unmount the volume.  This delay specifies the time between umount
+	 * retries.
+	 */
+
+	uint32_t udelay;
+
+	/* Interrupt related operations all hidden behind callbacks to isolate the
+	 * automounter from differences in interrupt handling by varying boards
+	 * and MCUs.  Board interrupts should be configured both insertion and
+	 * removal of the media can be detected.
+	 *
+	 * attach  - Attach or detach the media change interrupt handler to the
+	 *           board level interrupt
+	 * enable  - Enable or disable the media change interrupt
+	 */
+
+	CODE int (*attach)(FAR const struct automount_lower_s *lower,
+					  automount_handler_t isr, FAR void *arg);
+	CODE void (*enable)(FAR const struct automount_lower_s *lower,
+					  bool enable);
+	CODE bool (*inserted)(FAR const struct automount_lower_s *lower);
+};
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#ifdef __cplusplus
+#define EXTERN extern "C"
+extern "C" {
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: automount_initialize
+ *
+ * Description:
+ *   Configure the automounter.
+ *
+ * Input Parameters:
+ *   lower - Persistent board configuration data
+ *
+ * Returned Value:
+ *   A void* handle.  The only use for this handle is with automount_uninitialize().
+ *   NULL is returned on any failure.
+ *
+ ****************************************************************************/
+
+FAR void *automount_initialize(FAR const struct automount_lower_s *lower);
```

Comment:
We don't have fs_automount.c, Nuttx has.
Why do we (you) need this functionality?